SU2_CFD/include/solvers/CEulerSolver.hpp

```diff
@@ -37,7 +38,7 @@
  * a child class for each particular solver (Euler, Navier-Stokes, etc.)
  * \author F. Palacios
  */
-class CEulerSolver : public CSolver {
+class CEulerSolver : public CSolver, public DBMember<CEulerSolver, CSolverFactory> {
```

Comment:
What happens to CNSSolver? Which is derived from CEulerSolver and presumably will also need to inherit from DBMember<CNSSolver,...>.
Would it not be simpler to have a softer contract that solvers need to register themselves with some factory?.

How does this help with your goal of dynamic linking? Will you hard-code "USER_SOLVER" options into the factories?

Reply:
1. Every single other solver has to inherit from DBMember and provide GetName() and CreateMethod() to be registered automatically.
2. Concerning CNSSolver, I understand multiple inheritance may become confusing but it's irrelevant. Commit 37f62a2 virtually only adds those requirements to CNSSolver, and voila, it gets registered:
  - The knownSolvers map will only have 2 elements in this case: "EULER" (as CEulerSolver) and "NAVIER_STOKES" (as CNSSolver). But CNSSolver indirectly inherits from "DBMember<CEulerSolver, CSolverFactory>" so why it doesn't get registered as a CEulerSolver instead? because that only redefines CEulerSolver::is_registered (The static variable) and has nothing to do with the new solver registration :)  (I tested NS solver with the laminar flat plate case).

3. This doesn't help with dynamic linking yet; my end goal would be to take advantage of Global and shared/static Variables (instantiated well before loading libraries):
  - Macros will be helpful; mainly, "DeclareClassAsFactory" would initialize some global shared variables and "AddClassToFactory" (for example) would rewrite some of those variables and instantiate new ones to register the class
to the associated factory. This will enable automatic registration even for RunTime-loaded classes;  If it interests you, I can craft a simplified version to demonstrate the idea.